examples/DimensionReduction/emulate_sample_linlinexp.jl

```diff
@@ -0,0 +1,142 @@
+using Distributions
+using EnsembleKalmanProcesses
+using JLD2
+using LinearAlgebra
+using Random
+using MCMCChains
+using AdvancedMH
+
+# CES
+using CalibrateEmulateSample.Emulators
+using CalibrateEmulateSample.MarkovChainMonteCarlo
+using CalibrateEmulateSample.Utilities
+
+include("./models.jl")
+
+
+mutable struct NoEmulation <: Emulators.MachineLearningTool
+    f::Any
+    output_structure_mats::Any
+
+    NoEmulation(f) = new(f, nothing)
+end
+function Emulators.build_models!(ne::NoEmulation, iopairs, input_structure_mats, output_structure_mats; mlt_kwargs...)
+    ne.output_structure_mats = output_structure_mats
+end
+function Emulators.predict(ne::NoEmulation, new_inputs; mlt_kwargs...)
+    encoder_schedule = mlt_kwargs[:encoder_schedule]
+
+    decoded_inputs = decode_with_schedule(encoder_schedule, EnsembleKalmanProcesses.DataContainer(new_inputs), "in")
+    decoded_outputs = hcat(map(ne.f, eachcol(decoded_inputs.data))...)
+    encoded_outputs =
+        encode_with_schedule(encoder_schedule, EnsembleKalmanProcesses.DataContainer(decoded_outputs), "out").data
+
+    cov = Utilities.get_structure_mat(ne.output_structure_mats)
+    @assert cov ≈ cov'
+    (
+        encoded_outputs,
+        cat([Hermitian(cov) for _ in eachcol(new_inputs)]...; dims = 3),
+    )
+end
+
+
+rng = Random.MersenneTwister(41)
+input_dim = 100
+output_dim = 100
+αs = [0.0, 0.25, 0.5, 0.75, 1.0]
+
+num_trials = 1
+for trial in 1:num_trials
+    loaded = load("datafiles/ekp_linlinexp_$(trial).jld2")
```

Comment:
I found that loading this in a new session fails, it seems like the only time it worked was if I ran both `calibrate` and `emulate_sample` in the same session.